diff_drive_controller/src/diff_drive_controller.cpp

```diff
@@ -306,6 +323,99 @@ namespace diff_drive_controller
+    cmd_vel_limited_pub_->msg_.twist.angular.y = 0.0;
+
+    // Joint trajectory controller state:
+    state_pub_.reset(new realtime_tools::RealtimePublisher<DiffDriveControllerState>(controller_nh, "state", 100));
+
+    state_pub_->msg_.header.frame_id = base_frame_id_;
+
+    state_pub_->msg_.joint_names.resize(2 * wheel_joints_size_);
+
+    state_pub_->msg_.desired.positions.resize(2 * wheel_joints_size_);
+    state_pub_->msg_.desired.velocities.resize(2 * wheel_joints_size_);
+    state_pub_->msg_.desired.accelerations.resize(2 * wheel_joints_size_);
+    state_pub_->msg_.desired.effort.resize(2 * wheel_joints_size_);
+
+    state_pub_->msg_.actual.positions.resize(2 * wheel_joints_size_);
+    state_pub_->msg_.actual.velocities.resize(2 * wheel_joints_size_);
+    state_pub_->msg_.actual.accelerations.resize(2 * wheel_joints_size_);
+    state_pub_->msg_.actual.effort.resize(2 * wheel_joints_size_);
+
+    state_pub_->msg_.error.positions.resize(2 * wheel_joints_size_);
+    state_pub_->msg_.error.velocities.resize(2 * wheel_joints_size_);
+    state_pub_->msg_.error.accelerations.resize(2 * wheel_joints_size_);
+    state_pub_->msg_.error.effort.resize(2 * wheel_joints_size_);
+
+    state_pub_->msg_.actual_estimated.positions.resize(2 * wheel_joints_size_);
+    state_pub_->msg_.actual_estimated.velocities.resize(2 * wheel_joints_size_);
+    state_pub_->msg_.actual_estimated.accelerations.resize(2 * wheel_joints_size_);
+    state_pub_->msg_.actual_estimated.effort.resize(2 * wheel_joints_size_);
+
+    state_pub_->msg_.error_estimated.positions.resize(2 * wheel_joints_size_);
+    state_pub_->msg_.error_estimated.velocities.resize(2 * wheel_joints_size_);
+    state_pub_->msg_.error_estimated.accelerations.resize(2 * wheel_joints_size_);
+    state_pub_->msg_.error_estimated.effort.resize(2 * wheel_joints_size_);
+
+    state_pub_->msg_.actual_side_average.positions.resize(2);
+    state_pub_->msg_.actual_side_average.velocities.resize(2);
+    state_pub_->msg_.actual_side_average.accelerations.resize(2);
+    state_pub_->msg_.actual_side_average.effort.resize(2);
+
+    state_pub_->msg_.error_side_average.positions.resize(2);
+    state_pub_->msg_.error_side_average.velocities.resize(2);
+    state_pub_->msg_.error_side_average.accelerations.resize(2);
+    state_pub_->msg_.error_side_average.effort.resize(2);
+
+    state_pub_->msg_.actual_estimated_side_average.positions.resize(2);
+    state_pub_->msg_.actual_estimated_side_average.velocities.resize(2);
+    state_pub_->msg_.actual_estimated_side_average.accelerations.resize(2);
+    state_pub_->msg_.actual_estimated_side_average.effort.resize(2);
+
+    state_pub_->msg_.error_estimated_side_average.positions.resize(2);
+    state_pub_->msg_.error_estimated_side_average.velocities.resize(2);
+    state_pub_->msg_.error_estimated_side_average.accelerations.resize(2);
+    state_pub_->msg_.error_estimated_side_average.effort.resize(2);
```

Comment:
Can you be more specific? Do you mean a helper function for each block of 4 vectors?